drivers/i2c/i2c_mcux_lpi2c.c

```diff
@@ -105,6 +105,22 @@ static uint32_t mcux_lpi2c_convert_flags(int msg_flags)
 	return flags;
 }
 
+volatile int s_i;
+bool HAL_I2C_IsDeviceReady(LPI2C_Type *pI2C, uint16_t DevAddress,
+		uint32_t baudrate)
+{
+	bool isRdy = 0;
+	uint32_t safeDelay = 10000 * 150 / (baudrate / 1000);
+
+	LPI2C_MasterStart(pI2C, DevAddress, kLPI2C_Write);
+	for (s_i = 0; s_i < safeDelay; s_i++) {
```

Comment:
Can you replace this with `k_busy_wait()`?   It's gotta be a time-based delay, and an empty loop isn't the right way to do that.

Reply:
ok, Has updated